batch/batch/worker/worker.py

```diff
@@ -2423,26 +2456,46 @@ async def create_job(self, request):
             raise web.HTTPServiceUnavailable
         return await asyncio.shield(self.create_job_1(request))
 
-    async def get_job_log(self, request):
-        if not self.active:
-            raise web.HTTPServiceUnavailable
+    def _job_from_request(self, request):
         batch_id = int(request.match_info['batch_id'])
         job_id = int(request.match_info['job_id'])
         id = (batch_id, job_id)
         job = self.jobs.get(id)
         if not job:
             raise web.HTTPNotFound()
+        return job
+
+    async def get_job_log(self, request):
+        if not self.active:
+            raise web.HTTPServiceUnavailable
+        job = self._job_from_request(request)
         return web.json_response(await job.get_log())
 
+    async def get_job_resource_usage(self, request):
+        if not self.active:
+            raise web.HTTPServiceUnavailable
+        job = self._job_from_request(request)
+        resource_usage = await job.get_resource_usage()
+
+        boundary = '----WebKitFormBoundarywiBIWjWR7osAkgFI'
+
+        resp = web.StreamResponse(
+            status=200, reason='OK', headers={'Content-Type': f'multipart/mixed;boundary={boundary}'}
+        )
+        await resp.prepare(request)
+
+        with aiohttp.MultipartWriter('mixed', boundary=boundary) as mpwriter:
+            for task, data in resource_usage.items():
+                part = mpwriter.append(data)
+                part.set_content_disposition('attachment', filename=task)
+            await mpwriter.write(resp)
```

Comment:
I'm not really clued in on why we need multipart writer here, what's the reason for using it this way and not using a `FileResponse` or something?

Reply:
Does this not work? I feel like we should be able to just send a File as bytes and not do this ourselves.

Reply:
Also I think webkit is a safari thing, does that matter?